### ISSUE TITLE
Error while applying deletion_policy parameter for  google_sql_database resource

### ISSUE THREAD

**Issue by Anand-chenepalli** [terraform-google-modules/terraform-google-sql-db#435]
### TL;DR

When passing "Abandon" to database deletion policy or while removing the "abandon" deletion policy, getting empty error message during terraform apply.

  database_deletion_policy = "ABANDON"

│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.default[0],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 174, in resource "google_sql_database" "default":
│  174: resource "google_sql_database" "default" {
│
╵
╷
│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest2"],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 185, in resource "google_sql_database" "additional_databases":
│  185: resource "google_sql_database" "additional_databases" {
│
╵
╷
│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest3"],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 185, in resource "google_sql_database" "additional_databases":
│  185: resource "google_sql_database" "additional_databases" {

During second terraform apply , the error is not popping up.

### Expected behavior

Should able to apply or destroy the database_deletion_policy  parameter with out any error.

### Observed behavior

  # module.pgsql-db.module.gcp-pgsql.google_sql_database.default[0] will be updated in-place
  ~ resource "google_sql_database" "default" {
      ~ deletion_policy = "DELETE" -> "ABANDON"
        id              = "projects/xxxxxxxxxxxxxxxxxxxxxxxxx/instances/xxxxxxxxxxxxxxxxxxxxx/databases/testdevD"
        name            = "testdevDBX"
        # (5 unchanged attributes hidden)
    }

  # module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["xxxxxxxx"] will be updated in-place
  ~ resource "google_sql_database" "additional_databases" {
      ~ deletion_policy = "DELETE" -> "ABANDON"
        id              = "projects/xxxxxxxxxxxxxxx0f280/instances/xxxxxxxxxxxxxxxxxx/databases/xxxxxxxxxx"
        name            = "dbtest2"
        # (5 unchanged attributes hidden)
    }

module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest2"]: Modifying... [id=projects/xxxxxxxxxxxxxxxxxxxxxxxxx/instances/xxxxxxxxxxxxxxxxxxxxxxxxx/databases/dbtest2]
module.pgsql-db.module.gcp-pgsql.google_sql_database.default[0]: Modifying... [id=projects/xxxxxxxxxxxxxxxxxxxxxxxxx/instances/xxxxxxxxxxxxxxxxxxxxxxxxx/databases/testdevDBX]
module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest3"]: Modifying... [id=projects/xxxxxxxxxxxxxxxxxxxxxxxxx/instances/xxxxxxxxxxxxxxxxxxxxxxxxx/databases/dbtest3]
module.pgsql-db.module.gcp-pgsql.google_sql_user.default[0]: Modifying... [id=pg13-dev//xxxxxxxxxxxxxxxxxxxxxxxxx]

╷
│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.default[0],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 174, in resource "google_sql_database" "default":
│  174: resource "google_sql_database" "default" {
│
╵
╷
│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest2"],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 185, in resource "google_sql_database" "additional_databases":
│  185: resource "google_sql_database" "additional_databases" {
│
╵
╷
│ Error: 
│ 
│
│   with module.pgsql-db.module.gcp-pgsql.google_sql_database.additional_databases["dbtest3"],
│   on .terraform\modules\pgsql-db.gcp-pgsql\modules\postgresql\main.tf line 185, in resource "google_sql_database" "additional_databases":
│  185: resource "google_sql_database" "additional_databases" {

### Terraform Configuration

```hcl
database_deletion_policy = "ABANDON"
  user_deletion_policy     = "ABANDON"
```

### Terraform Version

```sh
Terraform v1.3.3
on windows_amd64
+ provider registry.terraform.io/hashicorp/google v4.56.0
+ provider registry.terraform.io/hashicorp/google-beta v4.56.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3

Your version of Terraform is out of date! The latest version
is 1.3.9. You can update by downloading from https://www.terraform.io/downloads.html
```

**Comment by bdashrad** [comment]
Also seeing this on module version 14.0.1

```
 $  terraform version
Terraform v1.3.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/external v2.2.3
+ provider registry.terraform.io/hashicorp/google v4.56.0
+ provider registry.terraform.io/hashicorp/google-beta v4.56.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.16.1
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
```
